File: JPABookNook/src/main/java/com/skilldistillery/booknook/entities/AnswerVote.java
package com.skilldistillery.booknook.entities;

import javax.persistence.EmbeddedId;
import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.MapsId;
import javax.persistence.Table;

@Entity
@Table(name = "answer_vote")
public class AnswerVote {
	
	@EmbeddedId
	private AnswerVoteId id;
	
	private Boolean vote;
	
	@ManyToOne
	@JoinColumn(name = "user_id")
	@MapsId(value = "userId")
	private User user;
	
	@ManyToOne
	@JoinColumn(name = "answer_id")
	@MapsId(value = "answerId")
	private Answer answer;

}

File: ngBookNook/src/app/models/answer-vote.ts
import { User } from './user';
export class AnswerVote {

 id: number;
 vote: boolean;
 user: User;
 answer: AnswerVote;

 constructor (
   id: number = 0,
   vote: boolean = false,
   user: User = {},
   answer: AnswerVote = {}
   ){
     this.id = id;
     this.vote = vote;
     this.user = user;
     this.answer = answer
  }
}
